test/validation/api/pktio/pktio.c

```diff
@@ -2100,6 +2100,61 @@ static void pktio_test_statistics_counters(void)
 	}
 }
 
+static void pktio_test_extra_stats(void)
+{
+	odp_pktio_t pktio;
+	odp_pktio_param_t pktio_param;
+	int num_info, num_stats, i;
+
+	odp_pktio_param_init(&pktio_param);
+	pktio_param.in_mode = ODP_PKTIN_MODE_SCHED;
+
+	pktio = create_pktio(0, ODP_PKTIN_MODE_DIRECT, ODP_PKTOUT_MODE_DIRECT);
+	CU_ASSERT_FATAL(pktio != ODP_PKTIO_INVALID)
+	CU_ASSERT_FATAL(odp_pktio_start(pktio) == 0);
+
+	num_info = odp_pktio_extra_stat_info(pktio, NULL, 0);
+	CU_ASSERT_FATAL(num_info >= 0);
+
+	num_stats = odp_pktio_extra_stats(pktio, NULL, 0);
+	CU_ASSERT_FATAL(num_stats >= 0);
+
+	CU_ASSERT_FATAL(num_info == num_stats);
+
+	/* No extra statistics supported */
+	if (num_stats == 0) {
+		CU_ASSERT(odp_pktio_stop(pktio) == 0);
+		CU_ASSERT(odp_pktio_close(pktio) == 0);
+		return;
+	}
+
+	odp_pktio_extra_stat_info_t stats_info[num_stats];
+	uint64_t extra_stats[num_stats];
+
+	num_info = odp_pktio_extra_stat_info(pktio, stats_info, num_stats);
+	CU_ASSERT_FATAL(num_info > 0);
+
+	num_stats = odp_pktio_extra_stats(pktio, extra_stats, num_stats);
+	CU_ASSERT_FATAL(num_stats > 0);
```

Comment:
Check may require that return value matches the previous return value (num_stats)

Reply:
Fixed in V5.